src/BitStringAddresses/bosefs.jl

```diff
@@ -328,14 +328,43 @@ function hopnextneighbour(b::BoseFS{N,M,A}, chosen) where {N,M,A<:BitString}
     end
     return BoseFS{N,M,A}(new_address), √prod
 end
-function hopnextneighbour(b::SingleComponentFockAddress, i)
+function hopnextneighbour(b::SingleComponentFockAddress, i; pitwisted::Bool = false, hardwall::Bool = false)
```

Comment:
The branching should be fine as `pitwisted` and `hardwall` are type parameters in `ExtendedHubbardReal1D`. Still it'd be good to check with benchmarks. If they do cause slowdowns, I'd suggest inlining the function and maybe passing the arguments as positional arguments instead of using `Val`s.

Reply:
Maybe you are right @mtsch. It should be tested, though.